eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/BatchingEventStorageEngine.java

```diff
@@ -237,6 +239,7 @@ private EventStreamSpliterator(Function<T, List<? extends T>> fetchFunction,
             super(Long.MAX_VALUE, NONNULL | ORDERED | DISTINCT | CONCURRENT);
             this.fetchFunction = fetchFunction;
             this.batchSize = batchSize;
+            this.smallResultTreshold = this.batchSize/2;
```

Comment:
What do you think about moving the setting of the `smallResultTreshold` to an actual method so that we have a little more control over it?

If I've followed the current implementation correctly enough, this feature is in essence replacing the `BatchingEventStorageEngine#fetchForAggregateUntilEmpty()` method.
This method was introduced to help people that do not have clean event stores (unlike yourself).
Hence why the field it sets is validated at the start of the `tryAdvance` method.

Having such a method would allow the batching implementation to be ambiguous about the outcome. 
In contrast, the JDBC implementation (for which you are making this adjustment) could override the result to half the batch size.
Added, I think this method should reside in the builder of the `BatchingEventStorageEngine`. 
This would greatly ease the configuration for users, as they no longer have to provide a custom implementation to override the method.

What do you think, @krosenvold?

Reply:
This would, by the way, mean that the `fetchForAggregateUntilEmpty` method should be deprecated in favor of the batch size.

From there, we can adjust the check to validate if the `smallResultTreshold` equals the `batchSize`. 
If it does, we're dealing with JPA and shouldn't impose this change.
If it does not equal the `batchSize`, we can assume JDBC''s override is in place (or the user's configuration, of course).
That would warrant the adjustment you've made to optimize the heuristics.